libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridGenerateCircularGrid.cpp

```diff
@@ -0,0 +1,233 @@
+    {
+        // Regular in x-direction
+
+        for (UInt m = 0; m < xValues.size(); ++m)
+        {
+            xValues[m] = blockSize * static_cast<double>(m);
+        }
+    }
+    else
+    {
+        // Graded in x-direction
+
+        const int size = parameters.num_columns + 1;
+        int muni = static_cast<int>(static_cast<double>(size) * parameters.uniform_columns_fraction);
+        muni += (size - 1 - muni) % 2;
+        const int minc = (size - muni) / 2 + 1;
+        const double alfm = std::pow(parameters.maximum_uniform_size_columns, 1.0 / static_cast<double>(minc));
+
+        double xValue = -blockSize * std::pow(alfm, static_cast<double>(minc));
+
+        auto computeDeltaX = [blockSize, alfm, muni, minc](const int m) -> double
+        {
+            double deltaX;
+
+            if (m < minc)
+            {
+                deltaX = blockSize * std::pow(alfm, static_cast<double>(minc - m));
+            }
+            else if (m < minc + muni)
+            {
+                deltaX = blockSize;
+            }
+            else
+            {
+                deltaX = blockSize * std::pow(alfm, static_cast<double>(m + 1 - minc - muni));
+            }
+
+            return deltaX;
+        };
+
+        for (int m = 0; m < static_cast<int>(xValues.size()); ++m)
+        {
+            xValue += computeDeltaX(m);
+            xValues[static_cast<UInt>(m)] = xValue;
+        }
+    }
+
+    return xValues;
+}
+
+std::vector<double> meshkernel::CurvilinearGridGenerateCircularGrid::ComputeYValues(const MakeGridParameters& parameters)
+{
+    std::vector<double> yValues(static_cast<UInt>(parameters.num_rows + 1));
+    double blockSize = parameters.block_size_y;
+
+    if (parameters.uniform_rows_fraction == 1.0 || parameters.maximum_uniform_size_rows == 1.0)
+    {
+        // Regular in y-direction
+
+        for (UInt n = 0; n < yValues.size(); ++n)
+        {
+            yValues[n] = blockSize * static_cast<double>(n);
+        }
+    }
+    else
+    {
+        // Graded in y-direction
+
+        int size = parameters.num_rows + 1;
+        int nuni = static_cast<int>(static_cast<double>(size) * parameters.uniform_rows_fraction);
+        nuni += (size - 1 - nuni) % 2;
+        int ninc = (size - nuni) / 2 + 1;
+        double alfn = std::pow(parameters.maximum_uniform_size_rows, 1.0 / static_cast<double>(ninc));
+        double yValue = -blockSize * std::pow(alfn, static_cast<double>(1 - nuni));
+
+        for (int n = 0; n < static_cast<int>(yValues.size()); ++n)
+        {
+            yValue += (n < nuni ? blockSize : blockSize * std::pow(alfn, static_cast<double>(n + 1 - nuni)));
+            yValues[static_cast<UInt>(n)] = yValue;
+        }
+    }
+
+    return yValues;
+}
+
+lin_alg::Matrix<meshkernel::Point> meshkernel::CurvilinearGridGenerateCircularGrid::GenerateRectangularGrid(const MakeGridParameters& parameters)
```

Comment:
i belive there is already a function for generating a rectangular grid

Reply:
The rectangular grid generated here is, usually slightly, different from the one generated by CurvilinearGridRectangularGrid. DeltaX and deltaY can vary with m and n

Reply:
So I have renamed the function to GeneratedGradedRectangularGrid